### ISSUE TITLE
Menu: Overlay is blocking click and contextmenu events

### ISSUE THREAD

**Issue by henev** [angular/material2#9320]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
When menu is open, any click event (including context menu events) outside of the menu and targeting clickable items, is expected to work from the first time. Menu should be closed along with the action handled by the clicked item.

#### What is the current behavior?
Overlay is blocking click events (context menu events). Clicking on the overlay first closes the menu and the user needs to click again to do his desired action (e.g. clicking navigation item that).

#### What are the steps to reproduce?
1. Go to documents/demo page.
1. Navigate to Components -> Menu
1. Click on the first example for demo
1. Try to click any other component page from the left navigation - Component page will open after second click attempt ( after the menu is closed ).

#### What is the use-case or motivation for changing an existing behavior?
The current functionality is a bad UX decision. Users usually expect to execute actions with one click (closing menu is counted towards the executed action). Making the user to click twice to close the menu and then proceed to the desired action is frustrating and not likely acceptable by most projects.

It also causes issues if developers want to do custom context menu with the Menu component.

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
* Angular 5.1.0
* Material 5.0.2

**Comment by Velzevul** [comment]
Agree. In my case I would like to get behaviour similar to that of Google Inbox:
<img width="377" alt="screenshot 2018-01-15 14 42 09" src="https://user-images.githubusercontent.com/1362615/34949192-b6ca7d90-fa0f-11e7-910e-d1470b9ae220.png">

In this example, I can open "more actions" menu but still interact with "quick-access" items with one click. 

A minor related issue is that overlay also prevents hover states on current items, which reveals the list of actions (again, see Google Inbox for example).

**Comment by utkant** [comment]
Any progress on this. I am having this problem on the auto-complete component as well. Have to click twice on item to get it selected.

**Comment by rshelnutt** [comment]
Baffling how such a glaring issue as this hasn't been addressed yet. Hours researching this and the closest solution I've come across is [using click simulation](https://stackoverflow.com/questions/3277369/how-to-simulate-a-click-by-using-x-y-coordinates-in-javascript#3277417). Anyone actually have something more concrete?

**Comment by texmax11** [comment]
If a backdrop isn't required in your case, I suggest you use the `hasBackdrop` directive on the `MatMenu` and set it to false

**Comment by gwenaellarmet** [comment]
Hello everyone, any news on this issue ? Or is the only solutions the hacky simulated click ?

**Comment by JonWallsten** [comment]
@jelbourn: I've been following this issue and the PR that was just merged for some time. We're using backdrop right now to prevent multiple menus being open at the same time. Today I disabled the backdrop and added the new `outsidePointerEvents` observer to the mat menu trigger to hide the current menu whenever an outside click is detected. It works fine for my use case, but not sure about side effects. 
I know this issue is for a menu WITH backdrop, but I guess it would be really hard to do this with the backdrop unless you remove pointer events from the backdrop, but then what's the point of having it?

I will open a PR with my fix when it's ready and we'll see if my approach is good enough.

**Comment by Ribosom** [comment]
For me the problem is that cdk-overlay-connected-position-bounding-box is preventing any clicks because it has to big dimensions.

I added following styles:
```sass
.cdk-overlay-connected-position-bounding-box
  width: min-content !important
  height: min-content !important
```

This is a hack and it would be better the underlying problem would be fixed (Tested with  "@angular/material": "14.2.5")

**Comment by albanx** [comment]
This issue is very annoying and bad UX. I have been monitoring user sessions, and every use after clicking on a Angular material dropdown, has to click twice the Search button (or any other clickable element) to take effect, looks like a glitch, like an issue, like a bad click.

I notice that the cdk-overlay-container covers all the screen causing this bug.
